src/chat.rs

```diff
@@ -1060,11 +1062,12 @@ impl Chat {
     }
 
     /// Returns true if user can send messages to this chat.
+    #[allow(clippy::nonminimal_bool)]
     pub async fn can_send(&self, context: &Context) -> Result<bool> {
         Ok(!self.id.is_special()
             && !self.is_device_talk()
-            && !self.is_mailing_list()
             && !self.is_contact_request()
+            && !(self.is_mailing_list() && self.param.get(Param::ListPost).is_none_or_empty())
             && self.is_self_in_chat(context).await?)
```

Comment:
Clippy wants me to change the condition to:
```rust
        Ok(!(self.id.is_special()
            || self.is_device_talk()
            || self.is_contact_request()
            || !self.is_self_in_chat(context).await?
            || (self.is_mailing_list() && self.param.get(Param::ListPost).is_none_or_empty())))
```

I personally like the current version better but we can also go with clippy's version if someone agrees with it.

Reply:
never argue with clippy to be honest.  it's like arguing with rustfmt, you might have a tiny argument but the consistency is not worth it.  so i strongly vote for removing the clippy exception :)

Reply:
Yes, clippy version is just as readable. You can also define `let cant_send = self.id.is_special() || ...` and return `Ok(!cant_send)` if you don't like the long `!(...)` expression.